src/firefly/js/templates/lightcurve/LcViewer.jsx

```diff
@@ -105,6 +105,12 @@ export class LcViewer extends PureComponent {
                     errorMsg = `The uploaded table is not valid. The ${getMissionName(converterId)} option requires frame_id, source_id, or both scan_id and frame_num.
                     Please select the "Other" upload option for tables that do not meet these requirements.`;
                 }
+
+                if (converterId === 'ptf') {
```

Comment:
Loi, this code has been introduced to deal with specific error and we said last we will cleanup and refactor the code but never get to it. I think we have time to do it and i propose to Lijun to go ahead and spend time to at least do this part and cleanup as much as possible, specially because we will use again for ZTF.

@lznakano , my proposal is the following: make use of {{converters.wise.isTableUploadValid}} function from the `LcConverterFactory.js`.
I've introduced this function in the past but is not used in the common code, only in `WiseMissionOption.js`. 

Among other cleanup that you might want to do and simply the code, I'd suggest at least to replace this piece of code:

```javascript
if (converterId === 'ptf') {
                    errorMsg = `The uploaded table is not valid. The ${getMissionName(converterId)} option requires pid.
                    Please select the "Other" upload option for tables that do not meet these requirements.`;
                }
```
with:

```js
errorMsg =  converterData.isTableUploadValid(converterId)
```
or something similar.

Right now, the function `isTableUploadValid` returns a boolean value but you can return a function with ```{errorMsg:'yada yada', isValid:boolean}``` and use it in `LcViewer.jsx` to show the error message if validation fails. 
Of course, you will need to add a specific function for PTF in `PTFMissionOptions.js`.

What do you think? @loitly @lznakano